crowbar_framework/app/models/service_object.rb

```diff
@@ -722,14 +725,14 @@ def proposal_create(params)
     # When we create a proposal, it might be "invalid", as some roles might be missing
     # This is OK, as the next step for the user is to add nodes to the roles
     # But we need to skip the after_save validations in the _proposal_update
-    _proposal_update(proposal, false)
+    _proposal_update(@bc_name, params["id"] || params[:name], proposal, false)
   end
 
   def proposal_edit(params, validate_after_save = true)
     params["id"] = "bc-#{@bc_name}-#{params["id"] || params[:name]}"
     proposal = {}.merge(params)
     clean_proposal(proposal)
-    _proposal_update(proposal, validate_after_save)
+    _proposal_update(@bc_name, params["id"] || params[:name], proposal, validate_after_save)
```

Comment:
> Maybe use a variable for `params["id"] || params[:name]` since it's used twice?

Again, this is a bug - only the original `params["id"]` is needed.